test/helpers/wrappers.go

```diff
@@ -57,13 +57,17 @@ func CurlFail(endpoint string, optionalValues ...interface{}) string {
 // only outputs the HTTP code returned by its execution against the specified
 // endpoint. It takes a variadic optinalValues argument. This is passed on to
 // fmt.Sprintf() and uses into the curl message
-func CurlWithHTTPCode(endpoint string, optionalValues ...interface{}) string {
+func CurlWithHTTPCode(method string, headers []string, endpoint string, optionalValues ...interface{}) string {
```

Comment:
Why did you add the headers options here? I think that there is a clear way: 
```
helpers.CurlWithHttpCode("http://%s/ -H 'test:1'", endpoint)
```

Having a new parameter with a header make things difficult to write, and with the option that I mention, anyone can extend Curl command without to much problems. 

As example: 
```
			dest := helpers.CurlFail("http://%s/public -X %s", srvIP[helpers.IPv4], method)
			destHeader := helpers.CurlFail("http://%s/public -H 'X-Test: True' -X %s",
				srvIP[helpers.IPv4], method)
````
from:
https://github.com/cilium/cilium/blob/master/test/runtime/Policies.go#L938-L940

Reply:
That's a good idea. I just need to pass the headers *before* the URL, but that should work. Thanks!

Reply:
Done.